### PR TITLE
KOGITO-5326 Setup deploy/promote notifications on demand

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5326

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/212
- https://github.com/kiegroup/kogito-runtimes/pull/1449
- https://github.com/kiegroup/optaplanner/pull/1436
- https://github.com/kiegroup/kogito-apps/pull/945
- https://github.com/kiegroup/kogito-examples/pull/800
- https://github.com/kiegroup/kogito-images/pull/602
- https://github.com/kiegroup/kogito-operator/pull/944
- https://github.com/kiegroup/rhpam-kogito-operator/pull/56

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by radtriste** [comment]
Jenkins retest this

File: Jenkinsfile.promote [hunk 1]
```diff
@@ -112,6 +117,16 @@ pipeline {
     }
 }
 
+void sendNotification() {
+    if (params.SEND_NOTIFICATION) {
+        emailext body: "**Deploy job** #${BUILD_NUMBER} was: ${currentBuild.currentResult}\nPlease look here: ${BUILD_URL}",
```
Comment (radtriste) [inline Jenkinsfile.promote]:
```suggestion
        emailext body: "**Promote job** #${BUILD_NUMBER} was: ${currentBuild.currentResult}\nPlease look here: ${BUILD_URL}",
```

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -227,6 +232,16 @@ pipeline {
     }
 }
 
+void sendNotification() {
+    if (params.SEND_NOTIFICATION) {
+        emailext body: "**Deploy job** #${BUILD_NUMBER} was: ${currentBuild.currentResult}\nPlease look here: ${BUILD_URL}",
+             subject: "[${getBuildBranch()}] Kogito examples",
```
Comment (radtriste) [inline Jenkinsfile.deploy]:
```suggestion
             subject: "[${getBuildBranch()}] Kogito Examples",
```

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**